web/src/components/MemoEditor.tsx

```diff
@@ -15,7 +15,7 @@ import "../less/memo-editor.less";
 
 const listItemSymbolList = ["- [ ] ", "- [x] ", "- [X] ", "* ", "- "];
 const emptyOlReg = /^(\d+)\. $/;
-const pairSymbols = ["[]", "()", '""', "''"];
+const pairSymbols = ["[]", "()", '""', "''", "{}", "<>", "``", "”“", "‘‘", "【】", "（）", "《》"];
```

Comment:
I don't think you'll want to have ``<>`` here. Purely because of maths.
Try typing the equation ``x < 1``, for example. That'll make it go ``x <>``
This would be an even bigger issue with the suggested deletion feature. I recomment dropping it.

Reply:
> I don't think you'll want to have `<>` here. Purely because of maths. Try typing the equation `x < 1`, for example. That'll make it go `x <>` This would be an even bigger issue with the suggested deletion feature. I recommend dropping it.

I think we can learn from vscode's

in `microsoft/vscode/extensions/xml/xml.language-configuration.json`
```xml
{
	"comments": {
		"blockComment": [ "<!--", "-->" ]
	},
	"brackets": [
		["<!--", "-->"],
		["<", ">"],
		["{", "}"],
		["(", ")"]
	],
	"autoClosingPairs": [
		{ "open": "{", "close": "}"},
		{ "open": "[", "close": "]"},
		{ "open": "(", "close": ")" },
		{ "open": "\"", "close": "\"", "notIn": ["string"] },
		{ "open": "'", "close": "'", "notIn": ["string"] },
		{ "open": "<!--", "close": "-->", "notIn": [ "comment", "string" ]},
		{ "open": "<![CDATA[", "close": "]]>", "notIn": [ "comment", "string" ]}
	],
	"surroundingPairs": [
		{ "open": "'", "close": "'" },
		{ "open": "\"", "close": "\"" },
		{ "open": "{", "close": "}"},
		{ "open": "[", "close": "]"},
		{ "open": "(", "close": ")" },
		{ "open": "<", "close": ">" }
	],
	"colorizedBracketPairs": [
	],
	"folding": {
		"markers": {
			"start": "^\\s*<!--\\s*#region\\b.*-->",
			"end": "^\\s*<!--\\s*#endregion\\b.*-->"
		}
	},
	"wordPattern": {
		"pattern": "(-?\\d*\\.\\d\\w*)|([^\\`\\~\\!\\@\\#\\$\\%\\^\\&\\*\\(\\)\\=\\+\\[\\{\\]\\}\\\\\\|\\;\\:\\'\\\"\\,\\.\\<\\>\\/\\?\\s]+)"
	}
}

```

Reply:
Uhh.. elaborate?

Reply:
Use the definition of surfingPairs for these five Pairs